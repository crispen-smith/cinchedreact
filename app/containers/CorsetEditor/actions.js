/*
 *
 * CorsetCreator actions
 *
 */

import actionTypes from './constants';

export function defaultAction() {
  return {
    type: actionTypes.default,
  };
}

export function create(corset) {
  return {
    type: actionTypes.create,
    corset,
  };
}

export function save() {
  return {
    type: actionTypes.save,
  };
}

export function isSaved() {
  return {
    type: actionTypes.isSaved,
  };
}

export function setDescription(description) {
  return {
    type: actionTypes.setDescription,
    description,
  };
}

export function setPrimaryImage(image) {
  // using an image object to allow for passing additional values
  return {
    type: actionTypes.setPrimaryImage,
    image,
  };
}

export function addImage(image) {
  // using an image object to allow for passing additional values
  return {
    type: actionTypes.addImage,
    image,
  };
}

export function setPrice(price) {
  return {
    type: actionTypes.setPrice,
    price,
  };
}

export function reset() {
  return {
    type: actionTypes.reset,
  };
}

export function publish() {
  return {
    type: actionTypes.publish,
  };
}

export function unpublish() {
  return {
    type: actionTypes.unpublish,
  };
}
