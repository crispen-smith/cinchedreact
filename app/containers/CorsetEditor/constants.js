/*
 *
 * CorsetCreator constants
 *
 */

import keymirror from 'keymirror';

const actionTypes = keymirror({
  default: null,
  save: null,
  isSaved: null,
  create: null,
  setDescription: null,
  setPrimaryImage: null,
  addImage: null,
  setPrice: null,
  publish: null,
  unpublish: null,
});

export default actionTypes;
