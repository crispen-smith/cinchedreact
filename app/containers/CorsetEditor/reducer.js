/*
 *
 * CorsetCreator reducer
 *
 */

import { fromJS } from 'immutable';
import actionTypes from './constants';

export const initialState = fromJS({});

function corsetCreatorReducer(state = initialState, action) {
  switch (action.type) {
    case actionTypes.save:
      return fromJS({
        ...state.toJS(),
        saving: true,
      });
    case actionTypes.isSaved:
      return setSaved(state);
    case actionTypes.setDescription:
      return setDescription(state, action);
    case actionTypes.setPrimaryImage:
      return setPrimaryImage(state, action);
    case actionTypes.addImage:
      return addImage(state, action);
    case actionTypes.setPrice:
      return setPrice(state, action);
    case actionTypes.publish:
      return publish(state, action);
    case actionTypes.unpublish:
      return unpublish(state, action);
    case actionTypes.default:
      return state;
    default:
      return state;
  }
}

/* eslint-disable no-underscore-dangle */
const __getCurrentCorset = state => {
  const { corsets } = state.toJS();
  const currentCorset = corsets.filter(
    corset => corset.name === state.toJS().currentCorset,
  );
  return currentCorset[0];
};

const __getNewCorsetList = (state, currentCorset) => {
  const corsetList = state.corsets.filter(
    corset => corset.name !== state.currentCorset,
  );

  corsetList.push(currentCorset);
  return corsetList;
};

/* eslint-enable */

const setDescription = (state, action) => {
  const newState = {
    ...state.toJS(),
  };
  const currentCorset = __getCurrentCorset(state);
  currentCorset.description = action.description;
  newState.corsets = __getNewCorsetList(newState, currentCorset);
  return fromJS(newState);
};

const setPrimaryImage = (state, action) => {
  const newState = {
    ...state.toJS(),
  };
  const currentCorset = __getCurrentCorset(state);
  currentCorset.primaryImage = action.image;
  newState.corsets = __getNewCorsetList(newState, currentCorset);
  return fromJS(newState);
};

const addImage = (state, action) => {
  const newState = {
    ...state.toJS(),
  };
  const currentCorset = __getCurrentCorset(state);
  if (!currentCorset.images) currentCorset.images = [];
  currentCorset.images.push(action.image);
  newState.corsets = __getNewCorsetList(newState, currentCorset);
  return fromJS(newState);
};

const setPrice = (state, action) => {
  const newState = {
    ...state.toJS(),
  };
  const currentCorset = __getCurrentCorset(state);
  currentCorset.price = action.price;
  newState.corsets = __getNewCorsetList(newState, currentCorset);
  return fromJS(newState);
};

const setSaved = state => {
  const newState = {
    ...state.toJS(),
    saving: false,
  };
  const currentCorset = __getCurrentCorset(state);
  currentCorset.saved = true;
  newState.corsets = __getNewCorsetList(newState, currentCorset);
  return fromJS(newState);
};

const publish = state => {
  const newState = {
    ...state.toJS(),
  };
  const currentCorset = __getCurrentCorset(state);
  currentCorset.published = true;
  newState.corsets = __getNewCorsetList(newState, currentCorset);
  return fromJS(newState);
};

const unpublish = state => {
  const newState = {
    ...state.toJS(),
  };
  const currentCorset = __getCurrentCorset(state);
  newState.corsets = __getNewCorsetList(newState, currentCorset);
  return fromJS(newState);
};

export default corsetCreatorReducer;
