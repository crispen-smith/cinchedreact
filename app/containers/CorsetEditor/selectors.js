import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the corsetEditor state domain
 */

const selectCorsetEditorDomain = state =>
  state.get('corsetEditor', initialState);

/**
 * Other specific selectors
 */

/**
 * Default selector used by CorsetEditor
 */

const makeSelectCorsetEditor = () =>
  createSelector(selectCorsetEditorDomain, substate => substate.toJS());

export default makeSelectCorsetEditor;
export { selectCorsetEditorDomain };
