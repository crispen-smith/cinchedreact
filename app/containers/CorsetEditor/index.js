/**
 *
 * CorsetCreator
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
// import { Helmet } from 'react-helmet';
import { createStructuredSelector } from 'reselect';

import { compose } from 'redux';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
// import {
//   makeSelectCorsetGallery,
//   makeSelectCurrentCorset,
//   makeSelectorFilteredCorsets,
// } from '../CorsetGallery/selectors';
import { makeSelectLoggedIn } from '../BrandedHeader/selectors';
import reducer from './reducer';
// import * as actions from './actions';
import saga from './saga';
// import FormComponent from '../../components/FormComponent';
// import NameBox from '../../components/NameBox';
// import DropDown from '../../components/DropDown';
// import SubmitButton from '../../components/SubmitButton';

/* eslint-disable react/prefer-stateless-function */
export class CorsetEditor extends React.Component {
  // constructor(props) {
  //   super(props);
  // }

  render() {
    if (this.props.loggedIn) {
      return <div>Placeholder</div>;
    }
    return <div>Placehoulder</div>;
  }
}

CorsetEditor.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
};

const mapStateToProps = createStructuredSelector({
  loggedIn: makeSelectLoggedIn(),
});

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'corsetGallery', reducer });
const withSaga = injectSaga({ key: 'corsetGallery', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(CorsetEditor);
