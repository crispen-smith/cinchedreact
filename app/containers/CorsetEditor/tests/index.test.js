import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import CorsetEditor from '../index';
import configureStore from '../../../configureStore';

describe('<CorsetEditor />', () => {
  let store;
  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('Renders a CorsetEditor', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <CorsetEditor />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
