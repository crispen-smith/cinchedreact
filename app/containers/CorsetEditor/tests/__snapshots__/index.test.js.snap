// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CorsetEditor /> Renders a CorsetEditor 1`] = `
<div>
  <form
    className="FormComponent__FC-jRyaDy hZINgg"
    onSubmit={[Function]}
  >
    PLACEHOLDER
  </form>
</div>
`;
